File: HackerRank/Data Structure/Linked Lists/Insert a node at the head of a linked list.java
static SinglyLinkedListNode insertNodeAtHead(SinglyLinkedListNode llist, int data) {
        SinglyLinkedListNode temp = llist;
        if(temp == null) {
            SinglyLinkedListNode new_node = new SinglyLinkedListNode(data);
            temp = new_node;
            return temp;
        }
        
        SinglyLinkedListNode new_node = new SinglyLinkedListNode(data);
        new_node.next = temp;
        return new_node;
        

}

File: HackerRank/Data Structure/Linked Lists/Insert a node at the head of a linked list.cpp
Node* Insert(Node *head,int data)
{
  // Complete this method
    Node*newnode=(Node*)malloc(sizeof(Node));
    newnode->data=data;
    
    if(head==NULL)
    {
        newnode->next=NULL;
        head=newnode;
    }
    else
    {
    Node*temp=head;
    newnode->next=head;
    
    }
    return newnode;
    
}